doc/api/typescript.md

```diff
@@ -52,7 +55,7 @@ added: v22.6.0
 
 > Stability: 1.1 - Active development
 
-The flag [`--experimental-strip-types`][] enables Node.js to run TypeScript
+The flag [`--no-experimental-strip-types`][] prevents Node.js from running TypeScript
 files. By default Node.js will execute only files that contain no
 TypeScript features that require transformation, such as enums or namespaces.
 Node.js will replace inline type annotations with whitespace,
```

Comment:
nit: I think we should specify the default behaviour before introducing the `--no-experimental-strip-types` flag to enhance clarity. 

Something like:  
```suggestion  
Node.js enables the execution of TypeScript files by default through a lightweight type stripping feature. This feature removes inline type annotations (replacing them with whitespace) without performing type checking or transformations that require JavaScript code generation, allowing Node.js to execute TypeScript code without requiring source maps. However, TypeScript features that require configuration settings from `tsconfig.json`, such as `paths` or syntax transformations, are intentionally unsupported.  

To prevent Node.js from running TypeScript files, use the `--no-experimental-strip-types` flag. With this flag enabled, Node.js will execute only files that do not contain any TypeScript syntax.  

For TypeScript features requiring transformation, such as enums or namespaces, use the `--experimental-transform-types` flag.  
```  

Additionally, at the beginning of this documentation, IMHO, we should clarify that one of the suggested approaches is enabled by default.

Reply:
CI is very flaky, I'll improve docs in a follow up PR